platform/lib/utils/sh.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+require('module-alias/register');
+const {spawn} = require('child_process');
+const {ROOT} = require('@lib/utils/project').paths;
+
+const DEFAULT_OPTIONS = {
+  workingDir: ROOT,
+};
+
+/**
+ * Executes a shell command.
+ *
+ * @param {string} the command string
+ * @param {string=''} an optional message being displayed after the command has
+ * terminated succesfully.
+ * @param {Object=DEFAULT_OPTIONS} an optional message being displayed after the command has
+ * terminated succesfully.
+ */
+function sh(string, ...params) {
```

Comment:
A story: I recently went through a process of wanting to exec a binary from NodeJs, so I discovered `spawn()`, and its options, and had the thought, hey, I can write a simple wrapper for this. Then over time, I added more and more code to handle stupid edge cases as I came across them…

Once I got to approximately the same amount of code as you have now I had the thought that someone must've already written a module to handle the simple case I was interested in, and discovered https://www.npmjs.com/package/execa. Maybe a module to keep in mind if this function starts getting out of hand.